tests/Backend/CommonPart1/BucketsTest.php

```diff
@@ -153,6 +153,12 @@ public function testBucketManipulation()
             'description' => 'this is just a test',
         ];
 
+        $testBucketId = $bucketData['stage'] . '.c-'.$bucketData['name'];
+
+        if ($this->_client->getBucket($testBucketId)) {
```

Comment:
Udělal bych z toho metodu toho testcasu - třeba `$this->ensureBucketNotExists($bucketId)` nebo možná líp `$this->dropBucketIfExists($bucketId)`. Na tohle narazíme ještě častěji u více testů.... a pak i u tabulek.

Reply:
Plus musíš ještě ošetřit situaci, kdy ten bucket neexistuje. To ti hodí exception.

Reply:
```
Keboola\StorageApi\ClientException : Bucket in.c-test not found
 /code/src/Keboola/StorageApi/Client.php:2099
 /code/src/Keboola/StorageApi/Client.php:1993
 /code/src/Keboola/StorageApi/Client.php:246
 /code/tests/Backend/CommonPart1/BucketsTest.php:158
```

Reply:
Tím spíš dává smysl ta extra metoda ;)

Reply:
> Udělal bych z toho metodu toho testcasu - třeba $this->ensureBucketNotExists($bucketId) nebo možná líp $this->dropBucketIfExists($bucketId). Na tohle narazíme ještě častěji u více testů.... a pak i u tabulek.

imho by na to šlo upravit StorageApiTestCase::initEmptyBucket

Reply:
No tam je este problem ze `StorageApiTestCase` este dedi `StorageApiSharingTestCase` a ta trieda ma tiez privatnu metodu `initEmptyBucket` a tam je ta implementacia odlisna pretoze je tam napr. parameter `backend`, ktory nieje v tom `StorageApiTestCase`  ale ak na tom trvas tak to upravim.

Reply:
To s tím vůbec nesouvisí IMHO. Ty potřebuješ aby ten bucket neexistoval. Ne aby ti ho to vytvořilo, ne?

Reply:
Jojo ja myslim, ze Erik to myslel tak, ze tam vyhodime to vytvaranie bucketov ako to tam je teraz ze si zavolas createBucket, a nahradime to tim `initEmptyBucket` a tak sa nemusime starat o to ci tam je alebo nie pretoze ta metoda to za teba vyriesi. Ale su tam dve rozne implementacie tejto metody pretoze teraz je to `private` takze by sme museli pridat `displayName` aby sme boli schopny otestovat to co potrebujeme ze pri zakladani bucketu mozeme nastavit iny `displayName` a potomupravit tu metodu `initEmptyBucket ` v `StorageApiSharingTestCase` co ale prinasa ten problem o ktorom pisem hore

Reply:
Ha, koukám dokonce, že je tam naimplemenované i \Keboola\StorageApi\Client::bucketExists 🤦‍♂ toho jsem si nevšimnul.

Reply:
Ajo to taky nie :/

Reply:
Ale stejně aspoň na jednom místě budeš potřebovat mít $client->createBucket a otestovat jak se to chová s prázdným displayName třeba. Takže budeš stejně potřebovat mazat ten bucket ručně...

Reply:
Asi by som to nekomplikoval a upravil to aby to pouzivalo `\Keboola\StorageApi\Client::bucketExists`.

Reply:
co ty na to @ErikZigo? btw. pridal som commit kde sa pouziva a metoda bucketExists

Reply:
On se nejaky testovaci bucket vytvari i v dalsich testech v ty classe. Tim ze vyhodis ten drop z konce si nejsem jisty jestli nerozbijes nejaky dalsi test. Zkousel jsi to?

Reply:
Joo jasne to ma nenapadlo ze vlastne je tam este jeden test co vytvara bucket s menom test, tak sa na to este zajtra pozriem teda.

Reply:
> Tim ze vyhodis ten drop z konce si nejsem jisty jestli nerozbijes nejaky dalsi test.

Pokud ano, tak ho stejně chceme opravit ;)

Ale bylo by ideální to zjistit dřív, než to mergneme, tak pustit build tagu přes noc?

Reply:
Jo tak to otagujem

Reply:
tak neviem ale ten tag tu na klientovy skoncil nejak divne https://travis-ci.com/github/keboola/storage-api-php-client/builds/153971719

Reply:
Tu som otagoval connection https://github.com/keboola/connection/pull/2420/files a testy presli tu https://travis-ci.com/github/keboola/connection/jobs/299586082

Reply:
Tak jestli testy prošli, tak super! :)